tasks/gov.js

```diff
@@ -47,21 +47,230 @@ task("gov:setPricingConfig")
     });
 
 
+task("gov:forkAccountsAndHealthScores", "Get all unique accounts that have entered an Euler market and their health scores")
+    .addPositionalParam("filename", "file name without .json suffix")
+    .setAction(async ({ filename }) => {
+        const fs = require("fs");
+
+        const {et, ctx, } = await setupGovernanceFork();
+
+        const transactions = await ctx.contracts.euler.queryFilter(
+            "EnterMarket",
+            "earliest",
+            "latest",
+        );
+
+        let result = [];
+        for (let i = 0; i < transactions.length; i++) {
+            let temp = {};
+            temp.account = transactions[i].args.account;
+            temp.underlying = transactions[i].args.underlying;
+            result.push(temp);
+        }
+
+        // unique event arguments based on accounts
+        const key = 'account';
+        const arrayUniqueByKey = [...new Map(result.map(item =>
+            [item[key], item])).values()];
+
+        // unique addresses regardless of markets 
+        const uniqueAddresses = [...new Set(arrayUniqueByKey.map(item => item.account))];
+
+        // compute health scores
+        let health_scores = {};
+        for (let account of uniqueAddresses) {
+            let detLiq = await ctx.contracts.exec.callStatic.detailedLiquidity(account);
+            let totalLiabilities = ethers.BigNumber.from(0);
+            let totalAssets = ethers.BigNumber.from(0);
+
+            for (let asset of detLiq) {
+                totalLiabilities = totalLiabilities.add(asset.status.liabilityValue);
+                totalAssets = totalAssets.add(asset.status.collateralValue);
+            }
+
+            let health = 0;
+            let violation = false;
+
+            if (totalAssets > 0) {
+                health = totalAssets.mul(et.c1e18);
+
+                if (totalLiabilities > 0) {
+                    health = totalAssets.mul(et.c1e18).div(totalLiabilities);
+                }
+
+                if (health.gte(et.c1e18)) {
+                    violation = false;
+                    console.log(`Account ${account} not in violation`);
+                } else {
+                    violation = true;
+                    console.log(`Account ${account} in violation`);
+                }
+
+                health_scores[account] = {
+                    health: ethers.utils.formatEther(health),
+                    violation: violation
```

Comment:
the `violation` flag is not used anywhere. is it needed?

Reply:
Not needed. Will remove